src/fields/fft_poisson_solver/fft/WrapRocDST.cpp

```diff
@@ -0,0 +1,190 @@
+
+        // setting the initial values to 0
+        // we don't set the expanded Fourier array, because it will be initialized by the FFT
+        dst_plan.m_expanded_position_array->setVal<amrex::RunOn::Device>(0.,
+            dst_plan.m_expanded_position_array->box(), 0,
+            dst_plan.m_expanded_position_array->nComp());
+
+        // check for type of expanded size, should be const size_t *
+        const amrex::IntVect& expanded_size = expanded_position_box.length();
+        const std::size_t lengths[] = {AMREX_D_DECL(std::size_t(expanded_size[0]),
+                                                    std::size_t(expanded_size[1]),
+                                                    std::size_t(expanded_size[2]))};
+
+        // Initialize fft_plan.m_plan with the vendor fft plan.
+        rocfft_status result;
+        result = rocfft_plan_create(&(dst_plan.m_plan), \
+                                    rocfft_placement_notinplace, \
+                                    rocfft_transform_type_real_forward, \
+                                    precision, \
+                                    dim, \
+                                    lengths, \
+                                    1, \
+                                    nullptr);
+
+        assert_rocfft_status("rocfft_plan_create", result);
+
+        // Store meta-data in dst_plan
+        dst_plan.m_position_array = position_array;
+        dst_plan.m_fourier_array = fourier_array;
+
+        return dst_plan;
+    }
+
+    void DestroyPlan (DSTplan& dst_plan)
+    {
+        rocfft_plan_destroy( dst_plan.m_plan );
+    }
+
+    //template<direction d>
+    void Execute (DSTplan& dst_plan){
+        HIPACE_PROFILE("AnyDST::Execute()");
+
+        // Expand in position space m_position_array -> m_expanded_position_array
+        ExpandR2R(*dst_plan.m_expanded_position_array, *dst_plan.m_position_array);
+
+        rocfft_status result;
+
+        rocfft_execution_info execinfo = NULL;
+        result = rocfft_execution_info_create(&execinfo);
+        assert_rocfft_status("rocfft_execution_info_create", result);
+
+        std::size_t buffersize = 0;
+        result = rocfft_plan_get_work_buffer_size(dst_plan.m_plan, &buffersize);
+        assert_rocfft_status("rocfft_plan_get_work_buffer_size", result);
+
+        result = rocfft_execution_info_set_stream(execinfo, amrex::Gpu::gpuStream());
+        assert_rocfft_status("rocfft_execution_info_set_stream", result);
+
+        // R2C FFT m_expanded_position_array -> m_expanded_fourier_array
+        // 2nd argument type still wrong, should be void*
+        // reinterpret_cast<AnyFFT::Complex*>(dst_plan.m_expanded_fourier_array->dataPtr()), //3rd arg
+        result = rocfft_execute(dst_plan.m_plan, \
+                                (void**)&(dst_plan.m_expanded_position_array), \
+                                (void**)&(dst_plan.m_expanded_fourier_array), \
+                                execinfo);
```

Comment:
I have taken the bodies of these functions from #584:
Please double-check and feel free to add more PRs.

Reply:
Style note: line continuations with `\` are not needed here.